packages/fx-core/src/plugins/solution/fx-solution/commonQuestions.ts

```diff
@@ -208,6 +210,42 @@ export async function askResourceGroupInfo(
   }
 }
 
+async function getLocations(
+  ctx: SolutionContext,
+  rmClient: ResourceManagementClient
+): Promise<string[] | undefined> {
+  const credential = await ctx.azureAccountProvider!.getAccountCredentialAsync();
+  let subscriptionClient = undefined;
+  if (credential) {
+    subscriptionClient = new SubscriptionClient(credential);
+  } else {
+    throw returnUserError(
+      new Error(`Failed to get azure credential`),
+      "Solution",
+      SolutionError.FailedToGetAzureCredential
+    );
+  }
+  const askSubRes = await ctx.azureAccountProvider!.getSelectedSubscription(true);
+  const listLocations = await subscriptionClient.subscriptions.listLocations(
+    askSubRes!.subscriptionId
+  );
+  const locations = listLocations.map((item) => item.displayName);
+  const providerData = await rmClient.providers.get(MsResources);
+  const resourceTypeData = providerData.resourceTypes?.find(
+    (rt) => rt.resourceType?.toLowerCase() === ResourceGroups.toLowerCase()
+  );
+  const resourceLocations = resourceTypeData?.locations;
+  const rgLocations = resourceLocations?.filter((item) => locations.includes(item));
+  if (!rgLocations || rgLocations.length == 0) {
+    throw returnUserError(
+      new Error(`Failed to list resource group locations`),
+      "Solution",
+      SolutionError.FailedToListResourceGroupLocation
+    );
+  }
+  return rgLocations;
```

Comment:
Since you have checked `!rgLocations` so the return type can be `string[]` instead of `string[] | undefined`. This can simplify the type assertion afterwards.

Reply:
Make sense, fixed.